src/NServiceBus.Core/Pipeline/Outgoing/InspectOutgoingMessageBehavior.cs

```diff
@@ -0,0 +1,16 @@
+﻿namespace NServiceBus
+{
+    using System;
+    using System.Threading.Tasks;
+    using NServiceBus.Pipeline;
+
+    class InspectOutgoingMessageBehavior : Behavior<IOutgoingPhysicalMessageContext>
```

Comment:
We need to audit what time the message being audited was sent, at least that's the semantics of that header in the past.

The right thing to do is probably to have a "OriginalSentTime" header but that is going to be nasty from a backwards compatibility perspective :(

Another reason to stop [attaching audit data to audit messages](https://github.com/Particular/NServiceBus/issues/3494)